tests/common-tests.h

```diff
@@ -3317,6 +3317,122 @@ TEST_CASE_METHOD(common_tests, "Dynamic binding with type conversions", "[core][
     }
 }
 
+// Dynamic bindings with type casts
+TEST_CASE_METHOD(common_tests, "Dynamic row binding 4", "[core][dynamic]")
+{
+    soci::session sql(backEndFactory_, connectString_);
+
+    SECTION("simple type cast")
+    {
+        auto_table_creator tableCreator(tc_.table_creator_1(sql));
+
+        sql << "insert into soci_test(id, d, str, tm)"
+            << " values(10, 20.0, 'foobar',"
+            << tc_.to_date_time("2005-12-19 22:14:17")
+            << ")";
+
+        {
+            row r;
+            sql << "select id from soci_test", into(r);
+
+            static_assert(std::numeric_limits<int32_t>::digits10 == 9, "");
```

Comment:
Sorry, could you please add a comment (or maybe message in the assert?) explaining why is this needed, because it doesn't seem obvious to me (i.e. I don't understand it)?

Reply:
Whoops, that's a leftover that I've missed to remove after testing.